epub33/core/index.html

```diff
@@ -10732,6 +10732,12 @@ <h3>Syntax</h3>
 					contain <a href="#viewport.ebnf.sep-char">separator characters</a> or the <a
 						href="#viewport.ebnf.assign">assignment character</a>.</p>
 
+				<p>
+					If the <code>height</code> or the <code>width</code> assignments have multiple occurences whithin an otherwise
+					correct value of the [^meta/content^] attribute, only the first occurence of the <code>height</code>, respectively
+					<code>width</code>, definition counts, and all others MUST be ignored.
```

Comment:
Isn't this statement in the wrong spec? Sounds like the author is not allowed to delete the duplicates? 😉 

At most you might note that reading systems will only process the first value, but it seems kind of pointless to let authors know how invalid syntax is handled. The publication is gone and broken at that point.

Reply:
To finish the thought, I'd expected this to be a reading system requirement like:

> Reading systems MUST use the first declaration of a `width` and `height` property in a `viewport meta` tag (i.e., ignore repeated declarations).

Reply:
The question is: is it syntactically ok to have multiple `height` clause in the viewport setting? At this moment, per the BNF, it actually is. The reason I put the statement where it is now is to make this explicit.

Another possibility is to make a statement in the core spec whereby there _SHOULD NOT_ be repeated values in the viewport specification, and expand the RS spec with the statement you propose. That may be a better choice indeed.

Reply:
I will change the text in both specs along these lines.

Reply:
>Another possibility is to make a statement in the core spec whereby there SHOULD NOT be repeated values

I'm fine with a statement that says properties MUST NOT be repeated for authoring. There's no valid need for allowing repetition that I can think of. RSes need to account for this possibility, but it shouldn't be an allowed authoring pattern.

Reply:
CC-ing to @rdeltour and @wareid. If we put a MUST NOT, this becomes an epubcheck issue and, theoretically, a backward compatibility issue (that is why I shied away from a MUST NOT, actually). Is it o.k. to put a MUST NOT? Do we think that we would invalidate content out there? Do we care because that would be invalid content already?

Reply:
From a reading system perspective (more specifically, Thorium / Readium), we use ad-hoc / laser-focused regular expressions to parse the `content` attribute value, based on "common sense" syntactical expectations (i.e. unique `width` and `height` key/value pairs). I have personally never encountered problematic multiple `width` or `height` declarations. I haven't formally tested the edge case either, but I suspect that my parsing code would break somewhere :)

Reply:
UPDATE: our parsing code works with multiple `width` / `height` key/value pairs, we just pick the first regular expression match. Phew! ;)

Reply:
Indeed, I have prepared some tests yesterday (https://github.com/w3c/epub-tests/pull/235) and I ran it through Thorium and the test worked 😀

Reply:
I'll have to ask around about what we're doing but I don't doubt it's entirely possible it's something similar to what @danielweck has described, since we favour getting content to work whenever possible. I think a MUST NOT will be good for publishers though, to catch any accidents in the production process. Even with our internal EPUBCheck scans, not every error is disqualifying.

Reply:
At the moment, epubcheck does not catch a duplicate w/h setting. If we put a _MUST NOT_, it eventually will. Won't this violate our charter restriction? The [Safari guide](https://developer.apple.com/library/archive/documentation/AppleApplications/Reference/SafariHTMLRef/Articles/MetaTags.html#//apple_ref/doc/uid/TP40008193-SW6) that was used in previous incarnations of EPUB, is also silent on the issue.

At this moment, to stay in line with the charter, I think I would prefer to keep it a _SHOULD NOT_. To my spite, to be honest, but violation of the charter could not be justified by some possible security problems, which was the case with some other cases.

Reply:
Whatever we decide can be implemented in EPUBCheck. In fact I changed the way the viewport is parsed, using dedicated parsing code instead of the naïve string matching used in v4.x.

@iherman wrt the concern about backward compatibility:
- a SHOULD will make EPUBCheck raise warnings which we know often means rejection by distributors.
- EPUBCheck v5 will still be backward incompatible due to the EBNF which is more restrictive to what we had before (EPUBCheck did not complain about `"anything, width=100, height=100` but will raise a syntax error in v5.0.0).

Reply:
I don't see how putting a "should not" in helps. That suggests there are valid cases for duplicating the dimensions.

We're not supposed to invalidate valid content, but in this case all we're doing is better defining what was always the only valid representation. That's not breaking our charter goal. I can't imagine someone would be upset that they're now being alerted to a potential incompatibility across reading systems.

I don't care if the ebnf allows properties without values, if that's really a concern. If it's to capture a blown height/width declaration, that's going to happen anyway. If it's to allow other garbage through, so be it.

Reply:
Ok. I changed it to _MUST NOT_.